controllers/cloud.redhat.com/makers/maker.go

```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
 	}}
 }
 
+func (m *Maker) makeJob(job crd.Job, app *crd.ClowdApp, hash string) error {
+
+	nn := types.NamespacedName{
+		Name:      fmt.Sprintf("%v-%v", app.Name, job.Name),
+		Namespace: m.Request.Namespace,
+	}
+
+	c := batch.CronJob{}
+	err := m.Client.Get(m.Ctx, nn, &c)
+
+	update, err := utils.UpdateOrErr(err)
+	if err != nil {
+		return err
+	}
+
+	initJob(m.App, m.Env, &c, nn, job, hash)
+
+	if err := update.Apply(m.Ctx, m.Client, &c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func initJob(app *crd.ClowdApp, env *crd.ClowdEnvironment, jc *batch.CronJob, nn types.NamespacedName, job crd.Job, hash string) {
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(jc, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := job.PodSpec
+
+	jc.Spec.Schedule = job.Schedule
+
+	jc.Spec.JobTemplate.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
+
+	jc.Spec.JobTemplate.Spec.Template.Spec.ImagePullSecrets = []core.LocalObjectReference{
+		{Name: "quay-cloudservices-pull"},
+	}
+
+	envvar := pod.Env
+	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
+
+	var livenessProbe core.Probe
+	var readinessProbe core.Probe
+
+	if pod.LivenessProbe != nil {
+		livenessProbe = *pod.LivenessProbe
+	} else {
+		livenessProbe = core.Probe{}
+	}
+	if pod.ReadinessProbe != nil {
+		readinessProbe = *pod.ReadinessProbe
+	} else {
+		readinessProbe = core.Probe{}
+	}
+
+	var lcpu, lmemory, rcpu, rmemory resource.Quantity
+	nullCPU := resource.Quantity{Format: resource.DecimalSI}
+	nullMemory := resource.Quantity{Format: resource.BinarySI}
+
+	if *pod.Resources.Limits.Cpu() != nullCPU {
```

Comment:
Can this block of code be reused between this and `initDeployment`?

Reply:
I'll DRY it up